src/net/sourceforge/kolmafia/persistence/DebugDatabase.java

```diff
@@ -2250,9 +2255,8 @@ private static void loadScrapeData(final StringArray array, final String fileNam
             currentHTML.append(currentLine);
             currentHTML.append(KoLConstants.LINE_BREAK);
           } while (!currentLine.equals("</html>"));
-
-          if (array.get(currentId).equals("")) {
-            array.set(currentId, currentHTML.toString());
+          if (stringMap.get(currentId) == null || stringMap.get(currentId).isEmpty()) {
```

Comment:
You could reduce this to a single hash table lookup via ```stringMap.getOrDefault(currentId, "").isEmpty()```

Reply:
I'm not convinced the latter case actually happens anymore! But yes I will do this.